pkg/controller/core/queuedworkload_controller.go

```diff
@@ -134,6 +141,7 @@ func (r *QueuedWorkloadReconciler) Delete(e event.DeleteEvent) bool {
 	if wl.Spec.Admission == nil {
 		r.queues.DeleteWorkload(wl)
 	}
+	r.notifyWatchers(wl)
```

Comment:
I am a bit confused here, why calling it before `return` if it has been deferred in line 107 already

Reply:
Different functions :)

Reply:
then why not defer here as well? just so they all look the same